src/routes/callback.ts

```diff
@@ -0,0 +1,66 @@
+import { Request, Response } from 'express';
+import querystring from 'querystring';
+import request from 'request';
+import { STATE_KEY, SPOTIFY_API_TOKEN } from '../constants';
+
+const {
+  SPOTIFY_CLIENT_ID,
+  SPOTIFY_CLIENT_SECRET,
+  SPOTIFY_REDIRECT_URI,
+} = process.env;
+const route = (req: Request, res: Response) => {
+  // your application requests refresh and access tokens
+  // after checking the state parameter
+
+  const code = req.query.code || null;
+  const state = req.query.state || null;
+  const storedState = req.cookies ? req.cookies[STATE_KEY] : null;
+
+  if (state === null || state !== storedState) {
+    const mismatch = querystring.stringify({ error: 'state_mismatch' });
+    res.redirect(`/#${mismatch}`);
+  } else {
+    res.clearCookie(STATE_KEY);
+    const authStr = `${SPOTIFY_CLIENT_ID}:${SPOTIFY_CLIENT_SECRET}`;
+    const authBuffer = Buffer.from(authStr).toString('base64');
+    const authOptions = {
+      url: SPOTIFY_API_TOKEN,
+      form: {
+        code,
+        redirect_uri: SPOTIFY_REDIRECT_URI,
+        grant_type: 'authorization_code',
+      },
+      headers: {
+        Authorization: `Basic ${authBuffer}`,
+      },
+      json: true,
+    };
+
+    request.post(authOptions, (error, response, body) => {
+      if (!error && response.statusCode === 200) {
+        /* eslint-disable camelcase */
+
+        const { access_token: accessToken, refresh_token: refreshToken } = body;
```

Comment:
You’re a ⭐️! This is awesome! Does this mean we can get rid of the eslint ignore above?

Reply:
Done